### ISSUE TITLE
Invalid escape sequence \. when calling install_basic_package_files

### ISSUE THREAD

**Issue by xEnVrE** [robotology/ycm#245]
**Description of the issue:**

I am trying to compile from source [this](https://github.com/robotology/superimpose-mesh-lib/tree/devel) package that requires `YCM 0.10.1`.

When I execute `cmake` I obtain the following error message

```
CMake Error at /home/user/robot-install/share/YCM/modules/InstallBasicPackageFiles.cmake:618 (if):
  Syntax error in cmake code at

    /home/user/robot-install/share/YCM/modules/InstallBasicPackageFiles.cmake:618

  when parsing string

    ^[0-9][0-9\.]*$

  Invalid escape sequence \.
Call Stack (most recent call first):
  CMakeLists.txt:92 (install_basic_package_files)
```
 I am using `Ubuntu 16.04` , `cmake 3.5.1` and last `master` branch of `ycm`.

**Comment by drdanz** [comment]
This is the code causing this.

https://github.com/robotology/ycm/blob/ae223b18347a641873ca40aff482d15526ab9cde/modules/InstallBasicPackageFiles.cmake#L614-L629

@xEnVrE What version of CMake are you using?

CC @claudiofantacci

**Comment by xEnVrE** [comment]
> What version of CMake are you using?

CMake 3.5.1

**Comment by drdanz** [comment]
Weird, I cannot reproduce it with CMake 3.5.1 on debian

**Comment by xEnVrE** [comment]
I have a fresh installation of `Ubuntu 16.04` somewhere. I will try again and let you know if it depends on my current configuration or if I get the same error again.

**Comment by PeterBowman** [comment]
I think you are missing a slash, i.e. use `\\.` instead `\.` (inspired by https://github.com/kokkos/kokkos/issues/1661). Quick test with CMake 3.5.1:

```cmake
cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
project(cmake-match LANGUAGES NONE)
set(str "2.3")

if(str MATCHES "^[0-9].[0-9]$")
    message(STATUS "Matches (1)")
else()
    message(STATUS "No match (1)")
endif()

if(str MATCHES "^[0-9]\.[0-9]$")
    message(STATUS "Matches (2)")
else()
    message(STATUS "No match (2)")
endif()

if(str MATCHES "^[0-9]\\.[0-9]")
    message(STATUS "Matches (3)")
else()
    message(STATUS "No match (3)")
endif()
```

Outputs:

```
Matches (1)
Matches (2)
Matches (2)
```

However, If I `set(str "253")`:

```
Matches (1)
Matches (2)
No match (3)
```

So, literals need to be double-escaped with a `\\`.

**Comment by drdanz** [comment]
Thanks @PeterBowman!
@xEnVrE can you please try replacing `\.` with `\\.` at lines 618 and 624?

**Comment by xEnVrE** [comment]
Thanks @PeterBowman.

@drdanz it works with the suggested changes!

**Comment by drdanz** [comment]
Thanks @xEnVrE and @PeterBowman!
Fixed in 41f9a1d735317c19b3ad5f911403792a0d8477fc.
